rskj-core/src/main/java/co/rsk/net/sync/DecidingSyncState.java

```diff
@@ -0,0 +1,42 @@
+package co.rsk.net.sync;
+
+import java.time.Duration;
+
+public class DecidingSyncState extends BaseSyncState {
+    private PeersInformation knownPeers;
+
+    public DecidingSyncState(SyncConfiguration syncConfiguration, SyncEventsHandler syncEventsHandler, SyncInformation syncInformation, PeersInformation knownPeers) {
+        super(syncInformation, syncEventsHandler, syncConfiguration);
+
+        this.knownPeers = knownPeers;
+    }
+
+    @Override
+    public boolean isSyncing() {
```

Comment:
if it's syncing or not is used by miners to know if they should mine, are we sure miners can't start mining even if we need to sync?

Reply:
The MinerImpl check doesn't depends on this implementation it should be aware of this when NodeBlockProcessor is reworked